pubsublite/internal/wire/resources.go

```diff
@@ -40,6 +41,109 @@ func ValidateRegion(input string) error {
 	return nil
 }
 
+// ZoneToRegion returns the region that the given zone is in.
+func ZoneToRegion(zone string) (string, error) {
+	if err := ValidateZone(zone); err != nil {
+		return "", err
+	}
+	return zone[0:strings.LastIndex(zone, "-")], nil
+}
+
+// LocationPath stores a path consisting of a project and zone.
+type LocationPath struct {
+	// A Google Cloud project. The project ID (e.g. "my-project") or the project
+	// number (e.g. "987654321") can be provided.
+	Project string
+
+	// A Google Cloud zone, for example "us-central1-a".
+	Zone string
+}
+
+func (l LocationPath) String() string {
+	return fmt.Sprintf("projects/%s/locations/%s", l.Project, l.Zone)
+}
+
+var locPathRE = regexp.MustCompile(`^projects/([^/]+)/locations/([^/]+)$`)
+
+// ParseLocationPath parses a project/location path.
+func ParseLocationPath(input string) (LocationPath, error) {
+	parts := locPathRE.FindStringSubmatch(input)
```

Comment:
I think we need to check for "nil" here. Does len(nil) return zero or something like that?

Reply:
Yes, len(nil) returns zero.